spec/integration/criteria/time_with_zone_spec.rb

```diff
@@ -4,29 +4,140 @@
+      end
+
+      let(:view_range) do
+        Agency.collection.find(updated_at: {'$gt' => query_time - 10.minutes, '$lt' => query_time + 10.minutes})
+      end
+
+      it 'finds the document' do
+        expect(view_lt.to_a).to eq([agency1.attributes, agency2.attributes])
+        expect(view_gt.to_a).to eq([agency2.attributes, agency3.attributes])
+        expect(view_range.to_a).to eq([agency2.attributes])
+      end
     end
 
-    let(:time_in_zone) { Time.now.in_time_zone(time_zone_name) }
+    context 'Mongo driver dynamic field' do
+
+      let(:view_lt) do
+        Agency.collection.find(dynamic_time: {'$lt' => query_time + 10.minutes})
+      end
+
+      let(:view_gt) do
+        Agency.collection.find(dynamic_time: {'$gt' => query_time - 10.minutes})
+      end
+
+      let(:view_range) do
+        Agency.collection.find(dynamic_time: {'$gt' => query_time - 10.minutes, '$lt' => query_time + 10.minutes})
+      end
 
-    let(:view_lt) do
-      Agency.collection.find(updated_at: {'$lt' => time_in_zone + 10.minutes})
+      it 'finds the document' do
+        expect(view_lt.to_a).to eq([agency1.attributes, agency2.attributes])
+        expect(view_gt.to_a).to eq([agency2.attributes, agency3.attributes])
+        expect(view_range.to_a).to eq([agency2.attributes])
+      end
     end
 
-    let(:view_gt) do
-      Agency.collection.find(updated_at: {'$gt' => time_in_zone - 10.minutes})
+    context 'Mongoid static field' do
+
+      let(:view_lt) do
+        Agency.all.lt(updated_at: query_time + 10.minutes)
+      end
+
+      let(:view_gt) do
+        Agency.all.gt(updated_at: query_time - 10.minutes)
+      end
+
+      let(:view_range) do
+        Agency.all.where(updated_at: (query_time - 10.minutes)..(query_time + 10.minutes))
+      end
+
+      it 'finds the document' do
+        expect(view_lt.pluck(:_id).sort).to eq([agency1, agency2].pluck(:_id).sort)
+        expect(view_gt.pluck(:_id).sort).to eq([agency2, agency3].pluck(:_id).sort)
+        expect(view_range.pluck(:_id).sort).to eq([agency2].pluck(:_id).sort)
+      end
     end
 
-    let!(:agency) { Agency.create!.reload }
+    context 'Mongoid dynamic field' do
+
+      let(:view_lt) do
+        Agency.all.lt(dynamic_time: query_time + 10.minutes)
+      end
+
+      let(:view_gt) do
+        Agency.all.gt(dynamic_time: query_time - 10.minutes)
+      end
+
+      let(:view_range) do
+        Agency.all.where(dynamic_time: (query_time - 10.minutes)..(query_time + 10.minutes))
+      end
+
+      it 'finds the document' do
+        expect(view_lt.pluck(:_id).sort).to eq([agency1, agency2].pluck(:_id).sort)
+        expect(view_gt.pluck(:_id).sort).to eq([agency2, agency3].pluck(:_id).sort)
+        expect(view_range.pluck(:_id).sort).to eq([agency2].pluck(:_id).sort)
+      end
+    end
+  end
+
+  context 'query with Time' do
+    let(:query_time) { now_utc }
+
+    context 'in a UTC time zone' do
```

Comment:
This context and the outer context on line 100 need to have their descriptions expanded to clarify which times they refer to, because the outer context is already defining a UTC time hence it's unclear why the outer context isn't "in UTC time zone".

If your intention is to have the query times in various time zones, then line 101 needs to be clarified as to why that time is already in UTC (and not simply `Time.now` for example) and this context and its siblings need to say something like "when query argument is a time in UTC time zone".

Reply:
Done > "when zone of queried time is UTC"